File: Time_And_Space/mergeSortedArray.java
import java.util.Scanner;
public class mergeSortedArray
{
    public static Scanner scn = new Scanner(System.in);
    public static int[] solution(int[] A, int[] B)
    {
        int n = A.length;
        int m = B.length;
        if(n == 0 || m == 0)
        {
            return n == 0 ? B : A;
        }
        int i = 0;
        int j = 0;
        int k = 0;
        int[] ans = new int[n + m];

        while(i < n && j < m)
        {
            if(A[i] < B[j])
            {
                ans[k++] = A[i++];
            }
            else
            {
                ans[k++] = B[j++];
            }
        }
        while(i < n)
        {
            ans[k++] = A[i++];
        }
        while(j < m)
        {
            ans[k++] = B[j++];
        }

        return ans;
    }

    public static void main(String[] args)
    {
        int n = scn.nextInt();
        int[] A = new int[n];
        
        for(int i = 0; i < n; i++)
        {
            A[i] = scn.nextInt();
        }

        int m = scn.nextInt();
        int[] B = new int[m];
        
        for(int i = 0; i < m; i++)
        {
            B[i] = scn.nextInt();
        }
        
        int[] merge = solution(A, B);
        
        for(int i = 0; i < n + m; i++)
        {
            System.out.print(merge[i] + " ");
        }
    }
}

File: Time_And_Space/mergeSortedArray.cpp
#include<iostream>
#include<vector>
using namespace std;

vector<int> solution(vector<int>& A, vector<int>& B)
{
    int n = A.size();
    int m = B.size();
    if(n == 0 || m == 0)
    {
        return n == 0 ? B : A;
    }
    int i = 0;
    int j = 0;
    int k = 0;
    vector<int> ans(n + m, 0);

    while(i < n && j < m)
    {
        if(A[i] < B[j])
        {
            ans[k++] = A[i++];
        }
        else
        {
            ans[k++] = B[j++];
        }
    }
    while(i < n)
    {
        ans[k++] = A[i++];
    }
    while(j < m)
    {
        ans[k++] = B[j++];
    }

    return ans;
}

int main()
{
    int n;

    cin >> n;

    vector<int> A(n);

    for(int i = 0; i < n; i++)
    {
        cin >> A[i];
    }
    int m;
    cin >> m;
    vector<int> B(m);
    for(int i = 0; i < m; i++)
    {
        cin >> B[i];
    }
    vector<int> merge = solution(A, B);
    for(int i = 0; i < n + m; i++)
    {
        cout << merge[i] << " ";
    }
    return 0;
}